client/scala/src/main/scala/ai/verta/blobs/dataset/Dataset.scala

```diff
@@ -53,4 +70,34 @@ object Dataset {
      component.size.get,
      versionId
    )
+
+   /** Analogous to Python's os.path.expanduser
+    *  From https://stackoverflow.com/questions/6803913/java-analogous-to-python-os-path-expanduser-os-path-expandvars
+    *  @param path path
+    *  @return path, but with (first occurence of) ~ replace with user's home directory
+    */
+   private[dataset] def expanduser(path: String) = path.replaceFirst("~", System.getProperty("user.home"))
+
+   /** Hash the file's content
+    *  From https://stackoverflow.com/questions/41642595/scala-file-hashing
+    *  @param path filepath
+    */
+   private[dataset] def hash(file: File, algorithm: String) = Try {
+     val buffer = new Array[Byte](BufferSize)
+     val messageDigest = MessageDigest.getInstance(algorithm)
+
+     val dis = new DigestInputStream(
+       new FileInputStream(file),
+       messageDigest
+     )
+
+     try {
+       while (dis.read(buffer) != -1) {}
+     } finally {
+       dis.close()
+     }
```

Comment:
Not that here you did the right thing to close the stream.

Reply:
Acknowledged.